Projects/Server/Mobiles/Mobile.cs

```diff
@@ -4764,19 +4772,7 @@ public virtual void Kill()
             }
         }
 
-        HairInfo hair = null;
-        if (m_Hair != null)
-        {
-            hair = new HairInfo(m_Hair.ItemID, m_Hair.Hue);
-        }
-
-        FacialHairInfo facialhair = null;
-        if (m_FacialHair != null)
-        {
-            facialhair = new FacialHairInfo(m_FacialHair.ItemID, m_FacialHair.Hue);
-        }
-
-        var c = CreateCorpseHandler?.Invoke(this, hair, facialhair, content, equip);
+        var c = CreateCorpseHandler?.Invoke(this, Hair, FacialHair, content, equip);
```

Comment:
@kamronbatman is this ok? or is it potentially a problem to not be re-creating the objects before passing them like this?

Reply:
I mean... it works.... but I wondered if there was a reason it had been done like this, related to the use of Invoke :)